### ISSUE TITLE
coalesce:1: scalar parameter arg created globally in function coalesce`

### ISSUE THREAD

**Issue by kud** [sorin-ionescu/prezto#1055]
Does it mean anything to you?

I've often got this. Mostly with git apparently.

**Comment by sorin-ionescu** [comment]
Yes, I think there is a conflict, between what I assume is the Git completion, and the [`coalesce`](https://github.com/sorin-ionescu/prezto/blob/master/modules/helper/init.zsh#L25) function, which, thus far, has been used in prompt themes. How do I reproduce it?

**Comment by kud** [comment]
I've got it (I think) when:

- git pull
- the pull isn't finished yet, I do git log

Or maybe 

- git pull
- the pull isn't finished yet, git checkout <tab>

**Comment by sorin-ionescu** [comment]
Are you running those commands in parallel?

**Comment by kud** [comment]
No. It happens when I'm too fast for the cli in fact. I already typed the next command when the previous one wasn't finished.

**Comment by sorin-ionescu** [comment]
What Zsh version are you running? It has never happened to me. I cannot reproduce it.

**Comment by sorin-ionescu** [comment]
It happened to me too. It seems to be a race condition.

**Comment by kud** [comment]
An example.

```
~/P/_/play-tv ❯❯❯ git add .                                                                                           feat.tv-bottom ✱
gi%                                                                                                                                    ~/P/_/play-tv ❯❯❯ git commit -m "WIP"                                                                                 feat.tv-bottom ✚
git re[feat.tv-bottom a14bf5d] WIP
 3 files changed, 5 insertions(+), 1 deletion(-)
~/P/_/play-tv ❯❯❯ git recoalesce:1: scalar parameter arg created globally in function coalesce
```

**Comment by kud** [comment]
`zsh 5.2 (x86_64-apple-darwin15.0.0)`

**Comment by sorin-ionescu** [comment]
`coalesce` is called [here](https://github.com/sorin-ionescu/prezto/blob/master/modules/prompt/functions/prompt_sorin_setup#L131) and implemented [here](https://github.com/sorin-ionescu/prezto/blob/master/modules/helper/init.zsh#L25-L31).

**Comment by sorin-ionescu** [comment]
Do you have `setopt WARN_CREATE_GLOBAL`? Execute `setopt | grep warn` to find out.

**Comment by kud** [comment]
When `setopt | grep warn`, nothing returns.

**Comment by sorin-ionescu** [comment]
Paste `function coalesce { print "${${(s: :)@}[1]}" }` in your terminal, or replace it in [its file](https://github.com/sorin-ionescu/prezto/blob/master/modules/helper/init.zsh#L25-L31), and report if it still happens.

**Comment by Limess** [comment]
I have the same issue. I tried to replace the `coalesce` function but the issue still occurs occasionally.

**Comment by wonbyte** [comment]
seeing the same issue. Is there any work around?

**Comment by wonbyte** [comment]
@sorin-ionescu using your above coalesce function seems to resolve the issue for me.

**Comment by belak** [comment]
There is a problem with the above fix as mentioned in https://github.com/prezto-inactive-community-fork/prezto/issues/23

If anyone has a substitute it would be very helpful.
